src/main/java/com/extendedclip/deluxemenus/menu/MenuItem.java

```diff
@@ -74,31 +75,33 @@ public ItemStack getItemStack(@NotNull final MenuHolder holder) {
 
         final int temporaryAmount = amount;
 
-        if (isHeadItem(lowercaseStringMaterial) && this.options.headType().isPresent()) {
-            itemStack = getItemFromHook(this.options.headType().get().getHookName(), holder.setPlaceholders(stringMaterial.substring(this.options.headType().get().getPrefix().length())))
-                    .orElseGet(() -> DeluxeMenus.getInstance().getHead().clone());
-        } else if (ItemUtils.isItemsAdderItem(lowercaseStringMaterial)) {
-            itemStack = getItemFromHook("itemsadder", holder.setPlaceholders(stringMaterial.substring(ITEMSADDER_PREFIX.length())))
-                    .orElseGet(() -> new ItemStack(Material.STONE, temporaryAmount));
-        } else if (ItemUtils.isOraxenItem(lowercaseStringMaterial)) {
-            itemStack = getItemFromHook("oraxen", holder.setPlaceholders(stringMaterial.substring(ORAXEN_PREFIX.length())))
-                    .orElseGet(() -> new ItemStack(Material.STONE, temporaryAmount));
-        } else if (ItemUtils.isMMOItemsItem(lowercaseStringMaterial)) {
-            itemStack = getItemFromHook("mmoitems", holder.setPlaceholders(stringMaterial.substring(MMOITEMS_PREFIX.length())))
-                    .orElseGet(() -> new ItemStack(Material.STONE, temporaryAmount));
-        } else if (ItemUtils.isWaterBottle(lowercaseStringMaterial)) {
+        final String finalMaterial = lowercaseStringMaterial;
+        final ItemHook pluginHook = DeluxeMenus.getInstance().getItemHooks().values()
+            .stream()
+            .filter(x -> finalMaterial.startsWith(x.getPrefix()))
+            .findFirst()
+            .orElse(null);
+
+        if (pluginHook != null) {
+            itemStack = pluginHook.getItem(stringMaterial.substring(pluginHook.getPrefix().length()));
+        }

```

Comment:
orElse and if statement can be simplified with findFirst().ifPresent()

Reply:
When using ifPresent, we would have to turn ItemStack into an atomic for it to function